cmake/Stdlib.cmake

```diff
@@ -1,33 +1,229 @@
-            stdlib.ispc | \"${Python3_EXECUTABLE}\" stdlib2cpp.py mask${mask}
-            > ${output}
-        DEPENDS stdlib.ispc
+        OUTPUT ${cpp}
+        COMMAND ${Python3_EXECUTABLE} bitcode2cpp.py ${bc} --type=stdlib --runtime=${bit} --os=${OS_UP} ${cpp}
+        DEPENDS bitcode2cpp.py ${bc}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
     )
-    set(${outputPath} ${output} PARENT_SCOPE)
-    set_source_files_properties(${outputPath} PROPERTIES GENERATED true)
+
+    set(tmp_list_bc ${STDLIB_BC_FILES})
+    list(APPEND tmp_list_bc ${bc})
+    set(STDLIB_BC_FILES ${tmp_list_bc} PARENT_SCOPE)
+
+    set(tmp_list_cpp ${STDLIB_CPP_FILES})
+    list(APPEND tmp_list_cpp ${cpp})
+    set(STDLIB_CPP_FILES ${tmp_list_cpp} PARENT_SCOPE)
 endfunction()
 
-function(generate_stdlib resultList)
-    foreach (m ${ARGN})
-        create_stdlib(${m} outputPath)
-        list(APPEND tmpList "${outputPath}")
-        if(MSVC)
-            # Group generated files inside Visual Studio
-            source_group("Generated Stdlib" FILES ${outputPath})
-        endif()
+function (generate_stdlibs_1 ispc_name)
+    list(APPEND os_list)
+    if (ISPC_WINDOWS_TARGET)
+        list(APPEND os_list "windows")
+    endif()
+    if (ISPC_UNIX_TARGET)
+        list(APPEND os_list "unix")
+    endif()
+
+    if (NOT os_list)
+        message(FATAL_ERROR "Windows or Linux target has to be enabled")
+    endif()
+
+    # "Regular" targets, targeting specific real ISA: sse/avx
+    if (X86_ENABLED)
+        foreach (target ${X86_TARGETS})
+            foreach (bit 32 64)
+                foreach (os ${os_list})
+                    stdlib_to_cpp(${ispc_name} ${target} ${bit} ${os})
+                endforeach()
+            endforeach()
+        endforeach()
+    endif()
+
+    # XE targets
+    if (XE_ENABLED)
+        foreach (target ${XE_TARGETS})
+            # No cross-compilation for XE targets
+            if (WIN32)
+                stdlib_to_cpp(${ispc_name} ${target} 64 windows)
+            elseif (APPLE)
+                # no xe support
+            else()
+                stdlib_to_cpp(${ispc_name} ${target} 64 unix)
+            endif()
+        endforeach()
+    endif()
+
+    # ARM targets
+    if (ARM_ENABLED)
+        foreach (os ${os_list})
+            foreach (target ${ARM_TARGETS})
+                if (${os} STREQUAL "windows")
+                    continue()
+                endif()
+                stdlib_to_cpp(${ispc_name} ${target} 32 ${os})
+            endforeach()
+            # Not all targets have 64bit
+            stdlib_to_cpp(${ispc_name} neon-i32x4 64 ${os})
+            stdlib_to_cpp(${ispc_name} neon-i32x8 64 ${os})
+        endforeach()
+    endif()
+
+    # WASM targets
+    if (WASM_ENABLED)
+        foreach (target ${WASM_TARGETS})
+            foreach (bit 32 64)
+                stdlib_to_cpp(${ispc_name} ${target} ${bit} web)
+            endforeach()
+        endforeach()
+    endif()
```

Comment:
OK, here were have a code duplication again that will be super hard to maintain. It would be cool to move it to separate CMake module and pass "stdlib_to_cpp" or "target_ll_to_cpp" as a function pointer :) which is of course not possible in CMake. But we can do something like this:
```
function(process_targets cpp_function_name ispc_name_or_target targets_list bits_list os_list)
                if (${function_name} STREQUAL "stdlib_to_cpp")
                    ${function_name}(${ispc_name_or_target} ${target} ${bit} ${os})
                elseif (${function_name} STREQUAL "target_ll_to_cpp")
                    ${function_name}(${target} ${bit} ${os})
                else()
endfunction()
```
What do you think?

Reply:
If you decide to eliminate code duplication, please do this in a separate PR and it'll hopefully reduce number of changes in CMake in this PR.

Reply:
> OK, here were have a code duplication again that will be super hard to maintain. It would be cool to move it to separate CMake module and pass "stdlib_to_cpp" or "target_ll_to_cpp" as a function pointer :) which is of course not possible in CMake. But we can do something like this:
> 
> ```
> function(process_targets cpp_function_name ispc_name_or_target targets_list bits_list os_list)
>                 if (${function_name} STREQUAL "stdlib_to_cpp")
>                     ${function_name}(${ispc_name_or_target} ${target} ${bit} ${os})
>                 elseif (${function_name} STREQUAL "target_ll_to_cpp")
>                     ${function_name}(${target} ${bit} ${os})
>                 else()
> endfunction()
> ```
> 
> What do you think?

Done.